### PR TITLE
Replace distutils with packaging for version

### LINKED ISSUE #16254: [Bug]: netdata uses obsoleted distutils module (removed since python 3.12)
### Bug description

netdata uses obsoleted distutils module

«Code that imports distutils will no longer work from Python 3.12.»

See https://peps.python.org/pep-0632/

### Expected behavior

Use other way to get module version

### Steps to reproduce

```
$ grep distutils collectors/python.d.plugin/python_modules/bases/FrameworkServices/UrlService.py 
from distutils.version import StrictVersion as version
```

### Installation method

from source

### System info

```shell
$ name -a; grep -HvE "^#|URL" /etc/*release
bash: name: команда не найдена
/etc/altlinux-release:ALT Regular Sisyphus
```

### Netdata build info

```shell
$ sudo netdata -W buildinfo
Packaging:
    Netdata Version ____________________________________________ : v1.43.0
```

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

Fixes #16254 

Replaces `distutils.version` with `packaging.version` as suggested in https://peps.python.org/pep-0632/

##### Test Plan

Make sure python.d.plugin continues to work when it uses the `UrlService`.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: collectors/python.d.plugin/python_modules/bases/FrameworkServices/UrlService.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import urllib3
 
-from distutils.version import StrictVersion as version
+from packaging.version import parse as version
```
Comment (ilyam8) [inline collectors/python.d.plugin/python_modules/bases/FrameworkServices/UrlService.py:9]: Is it the standard library? Looks like it is not.

```python
~ $ python3
Python 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from packaging.version import parse as version
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'packaging'
```

Comment (MrZammler) [reply collectors/python.d.plugin/python_modules/bases/FrameworkServices/UrlService.py:9]: We could just remove it completely.

We use it to check the `urllib3` version, and we require a version >1.9. Considering this was released in 2014, we could assume by now we should be safe...

Comment (ilyam8) [reply collectors/python.d.plugin/python_modules/bases/FrameworkServices/UrlService.py:9]: Yes, I see 2 options:
 - remove it
 - copy StrictVersion from distutils to python_modules/third-party or something.

File: collectors/python.d.plugin/python_modules/bases/FrameworkServices/UrlService.py [hunk 2]
```diff
@@ -6,37 +6,18 @@
...
 try:
     urllib3.disable_warnings()
 except AttributeError:
     pass
 
-# https://github.com/urllib3/urllib3/blob/master/CHANGES.rst#19-2014-07-04
-# New retry logic and urllib3.util.retry.Retry configuration object. (Issue https://github.com/urllib3/urllib3/pull/326)
-URLLIB3_MIN_REQUIRED_VERSION = '1.9'
 URLLIB3_VERSION = urllib3.__version__
 URLLIB3 = 'urllib3'
```
[the remark this answers is not in the document]
Comment (MrZammler) [reply collectors/python.d.plugin/python_modules/bases/FrameworkServices/UrlService.py:17]: We use them to print the version a few lines below.

**Review by ilyam8 — APPROVED** [review]

**Review by tkatsoulas — APPROVED** [review]

**Pull request merged**